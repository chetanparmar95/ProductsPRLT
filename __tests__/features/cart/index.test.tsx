
import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CartScreen from '../../../src/features/cart';

const mockStore = configureStore([]);

describe('CartScreen Component', () => {
  it('CartScreen renders correctly', () => {
    const store = mockStore({
        products: {
          cartItems: [],
        },
    });
    const { toJSON } = render(
        <Provider store={store}>
            <CartScreen />
        </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders CartScreen with empty cart', () => {
    const store = mockStore({
      products: {
        cartItems: [],
      },
    });

    const { getByText } = render(
      <Provider store={store}>
        <CartScreen />
      </Provider>
    );

    const emptyCartText = getByText('Your cart is empty');
    expect(emptyCartText).toBeDefined();
  });

  it('renders CartScreen with cart items', () => {
    const store = mockStore({
      products: {
        cartItems: [
          {
            id: 1,
            name: 'Product 1',
            price: 10.0,
            quantity: 2,
            colour: "Black",
            img: "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024"
          },
          {
            id: 2,
            name: 'Product 2',
            price: 15.0,
            quantity: 1,
            colour: "Black",
            img: "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024"
          },
        ],
      },
    });

    const { getByText } = render(
      <Provider store={store}>
        <CartScreen />
      </Provider>
    );

    const product1Text = getByText('Product 1');
    const product2Text = getByText('Product 2');
    const totalText = getByText('Total: $35.00');

    expect(product1Text).toBeDefined();
    expect(product2Text).toBeDefined();
    expect(totalText).toBeDefined();
  });
})
