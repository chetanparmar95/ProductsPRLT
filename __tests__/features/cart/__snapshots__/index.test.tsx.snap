// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartScreen Component CartScreen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 16,
        }
      }
    >
      Your cart is empty
    </Text>
    <Text
      style={
        {
          "color": "#888",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Start shopping to add items to your cart.
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Total: $
      0.00
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FF5722",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontWeight": "bold",
              "marginLeft": 5,
            },
            {},
          ]
        }
      >
        Checkout
      </Text>
    </View>
  </View>
</View>
`;
